website/docs/components/code-block/partials/code/how-to-use.md

```diff
@@ -0,0 +1,126 @@
+## How to use this component
+
+The basic invocation requires a `@value` argument. The component encodes this argument before displaying it.
+
+```handlebars
+<Hds::CodeBlock
+  @value="aws ec2 --region us-west-1 accept-vpc-peering-connection"
+/>
+```
+
+### Title and description
+
+Optionally, you can pass a title and/or a description.
+
+```handlebars
+<Hds::CodeBlock
+  @language="bash"
+  @value="aws ec2 --region us-west-1 accept-vpc-peering-connection"
+as |CB|>
+  <CB.Title>
+    CodeBlock title
+  </CB.Title>
+  <CB.Description>
+    CodeBlock description
+  </CB.Description>
+</Hds::CodeBlock>
+```
+
+### Language
+
+The `language` argument sets the syntax highlighting used. We only support the following languages: `bash`, `go`, `hcl`, `json`, `log`, `shell-session`, and `yaml`. If you need additional languages <LinkTo class="doc-link-generic" @route="show" @model="about/support">contact the Design Systems Team</LinkTo>
+
+```handlebars
+<Hds::CodeBlock
+  @language="go"
+  @value="package main
+import fmt
+func main() {
+  fmt.Println(helloWorld)
+}"
+/>
+```
+
+### Copy button
+
+Set `hasCopyButton` to `true` to display a button for users to copy `CodeBlock` content to their computer clipboard.
+
+```handlebars
+<Hds::CodeBlock
+  @language="javascript"
+  @hasCopyButton={{true}}
+  @value="let codeLang=`JavaScript`;
+console.log(`I am ${codeLang} code`);"
+/>
+```
+
+### Line numbers
+
+Line numbers are displayed by default. Set `hasLineNumbers` to `false` to hide them.
+
+```handlebars
+<Hds::CodeBlock
+  @language="javascript"
+  @hasLineNumbers={{false}}
+  @value="let codeLang=`JavaScript`;
+console.log(`I am ${codeLang} code`);"
+/>
+```
+
+### Line wrapping
+
+By default, long lines of code will overflow the `CodeBlock` container requiring users to scroll to view the full content. Setting `hasLineWrapping` to `true` will wrap long lines of code instead.
+
+```handlebars
```

Comment:
something strange happens in this example, that doesn't happen in the showcase:
<img width="878" alt="screenshot_3225" src="https://github.com/hashicorp/design-system/assets/686239/82db3e48-9f6c-4765-b373-2fe29f257055">

(I copied the same snippet in the showcase, and it rendered correctly 🤷‍♂️)

Reply:
Looks like we could add `bottom: 0;` to `.hds-code-block.line-numbers .line-numbers-rows` styles.

Reply:
@KristinLBradley's solution seems to be resolving this. I'll re-create the example in the component PR and address it there.

Reply:
The above suggestion fixes this example but creates issues in a scrollable container, see screen-shot below.

<img width="525" alt="Screenshot 2023-10-31 at 11 58 40" src="https://github.com/hashicorp/design-system/assets/788096/91789aa9-34e6-4317-b0eb-7a5049331567">

I will investigate further in the component PR, but doesn't seem trivial. As with the other plugin fix, if you trigger a resize it will render correctly.

Reply:
Try `min-height: 100%;` instead.

Reply:
Thanks for your suggestion, Kristin! I added this in [b2ffbbb](https://github.com/hashicorp/design-system/pull/1687/commits/b2ffbbbbbabfbd523f6692f121933084085bb343)